src/python/google_cloud_utils/big_query.py

```diff
@@ -379,8 +379,18 @@ def insert(self, inserts):
       time.sleep(1)
 
       if not result:
+        # Use result from the first batch, appending errors from the rest.
         result = response
       else:
-        result['insertErrors'].extend(response['insertErrors'])
+        # If there are new errors from the current batch, append to the result.
+        new_errors = response.get('insertErrors')
+        if not new_errors:
+          continue
+
+        # Apparently result may not have errors, be careful.
+        if result.get('insertErrors'):
+          result['insertErrors'].extend(new_errors)
+        else:
+          result = response
```

Comment:
Why do we overwrite the result here ?

Reply:
let me give a try to upgrading it

Reply:
doesn't seem like we can use the version that has it fixed (`2.3.0`: https://github.com/PyCQA/pylint/issues/2767), but I've used `{}` instead of `None` and now it's happy